libsolidity/ast/Types.cpp

```diff
@@ -1800,6 +1816,33 @@ FunctionType::FunctionType(EventDefinition const& _event):
 	swap(paramNames, m_parameterNames);
 }
 
+FunctionType::FunctionType(FunctionTypeName const& _typeName):
+	m_location(_typeName.visibility() == VariableDeclaration::Visibility::External ? Location::External : Location::Internal),
+	m_isConstant(_typeName.isDeclaredConst()),
+	m_isPayable(_typeName.isPayable())
+{
+	if (_typeName.isPayable())
+		solAssert(m_location == Location::External, "Internal payable function type used.");
```

Comment:
That's why I really would like to combine those flags into a single enum or at least a single class that has internal checks for these restrictions.